src/core/observable.ts

```diff
@@ -1,10 +1,58 @@
-import {IDerivation, notifyDependencyReady, notifyDependencyStale} from "./derivation";
+import {IDerivation, changeDependenciesState} from "./derivation";
 import {globalState} from "./globalstate";
-import {SimpleSet} from "../utils/set";
+
+export class DerivationsSets {
+	size: number = 0;
+	data0 = {};
```

Comment:
I wouldn't be surprised, it is interesting to test that, that keeping one collection for all states and then filtering wherever the sets are used is faster? because `delete` is quite expensive and usually these sets are pretty constant over the lifetime of a derivation, so that would a lot of moving around.